Remora.Results/Result.cs

```diff
@@ -32,6 +32,17 @@ namespace Remora.Results;
 [PublicAPI]
 public readonly struct Result : IResult
 {
+    /// <summary>
+    /// Gets a successful result.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// This property returns a result whose <see cref="Error"/> property is set to null, therefore <see cref="IsSuccess"/> returns true.
+    /// The <see cref="Inner"/> property is also set to null.
+    /// </para>
+    /// </remarks>
+    public static IResult SuccessfulResult { get; } = FromSuccess();
```

Comment:
I'd shorten this to `Success` since it's a static property and it'll always be called with a reference to the `Result` type.

Consider adding a matching property to the typed result struct as well for symmetry.

Reply:
Shortened, added a matching property to the typed result aswell.